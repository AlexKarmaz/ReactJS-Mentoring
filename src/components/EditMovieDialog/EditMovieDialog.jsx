import React, {useCallback} from 'react';
import ModalDialog from '../Common/ModalDialog';
import MovieForm from '../Common/MovieForm'
import { connect } from 'react-redux'
import {movieActions, commonActions} from '../../store/actions'

const EditMovieDialog = ({onClose, movieForEdit, onConfirm, genres}) => {
    const onDialogClose = useCallback(() => onClose(), [onClose]);
    const onSubmit = useCallback((data) => onConfirm(data), [onConfirm]);

    return (
        <ModalDialog dialogTitle='Edit movie' onDialogClose={onDialogClose}>
            <MovieForm
                onSubmit={onSubmit}
                movieData={movieForEdit}
                genres={genres}
            />
        </ModalDialog>
    );
};

const mapStateToProps = (state) => ({
    movieForEdit: state.commonData.movieForEdit,
    genres: state.moviesData.genres,
});

const mapDispatchToProps = (dispatch) => ({
    onClose: () => dispatch(commonActions.closeEditMovieDialog()),
    onConfirm: (data) => dispatch(movieActions.editMovie(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditMovieDialog);
