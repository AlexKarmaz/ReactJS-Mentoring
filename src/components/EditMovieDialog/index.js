import EditMovieDialog from './EditMovieDialog';

export default EditMovieDialog;
