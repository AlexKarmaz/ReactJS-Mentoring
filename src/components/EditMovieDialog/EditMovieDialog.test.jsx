import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import EditMovieDialog from './EditMovieDialog';

describe('EditMovieDialog', () => {
  test('should be rendered correctly', () => {
    const initialState = {
        moviesData: {
          genres: ['Comedy', 'Drama', 'Horror'],
        },
        commonData: {
            movieForEdit: {}
        }
      };
    const store = configureStore([])(initialState);
    const form = mount(
      <Provider store={store}>
        <EditMovieDialog />
      </Provider>,
    );

    expect(form.find('.modalDialog-title').text()).toEqual('Edit movie');
    expect(form).toMatchSnapshot();
  });
});
