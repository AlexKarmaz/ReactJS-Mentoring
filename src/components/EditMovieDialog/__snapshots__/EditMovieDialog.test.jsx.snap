// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMovieDialog should be rendered correctly 1`] = `ReactWrapper {}`;
